opensrp-anc/src/main/java/org/smartregister/anc/library/util/Utils.java

```diff
@@ -420,19 +420,47 @@ public static ButtonAlertStatus getButtonAlertStatus(Map<String, String> details
         nextContactDate =
                 StringUtils.isNotBlank(nextContactDate) ? Utils.reverseHyphenSeperatedValues(nextContactDate, "/") : null;
 
-        buttonAlertStatus1.buttonText = String.format(textTemplate, nextContact, (nextContactDate != null ? nextContactDate :
+        buttonAlertStatus.buttonText = String.format(getDisplayTemplate(context, alertStatus, isProfile), nextContact, (nextContactDate != null ? nextContactDate :
                 Utils.convertDateFormat(Calendar.getInstance().getTime(), Utils.CONTACT_DF)));
 
+        alertStatus =
+                Utils.processContactDoneToday(details.get(DBConstantsUtils.KeyUtils.LAST_CONTACT_RECORD_DATE), alertStatus);
 
-        buttonAlertStatus =
-                Utils.processContactDoneToday(details.get(DBConstantsUtils.KeyUtils.LAST_CONTACT_RECORD_DATE), buttonAlertStatus);
+        buttonAlertStatus.buttonAlertStatus = alertStatus;
+        buttonAlertStatus.gestationAge = gestationAge;
+        buttonAlertStatus.nextContact = nextContact;
+        buttonAlertStatus.nextContactDate = nextContactDate;
 
-        buttonAlertStatus1.buttonAlertStatus = buttonAlertStatus;
-        buttonAlertStatus1.gestationAge = gestationAge;
-        buttonAlertStatus1.nextContact = nextContact;
-        buttonAlertStatus1.nextContactDate = nextContactDate;
+        return buttonAlertStatus;
+    }
 
-        return buttonAlertStatus1;
+    private static String getDisplayTemplate(Context context, String alertStatus, boolean isProfile) {
+        String displayTemplate;
+        if (StringUtils.isNotBlank(alertStatus) && !isProfile) {
+            switch (alertStatus) {
+                case ConstantsUtils.AlertStatusUtils.IN_PROGRESS:
+                    displayTemplate = context.getString(R.string.contact_in_progress);
+                    break;
+                case ConstantsUtils.AlertStatusUtils.NOT_DUE:
+                    displayTemplate = context.getString(R.string.contact_number_due);
+                    break;
+                default:
+                    displayTemplate = context.getString(R.string.contact_weeks);
+                    break;
+            }
+        } else {
+            switch (alertStatus) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A switch statement does not contain a break](https://app.codacy.com/app/OpenSRP/opensrp-client-anc/pullRequest?prid=4623153)